### ISSUE TITLE
Save/load database to/from json file

### ISSUE THREAD

**Issue by pierstitus** [rwweb/webmemex#1]
In order to reduce the chance of data loss and to get a better overview of what is actually stored it would be nice to have a possibility to store the local storage database as a JSON file. And ideally also being able to load it again.

I guess there might be browser ways/extensions to do these things, but I couldn't find a way quickly, apart from viewing the PouchDB in the Chromium Developer Tools.

**Comment by Treora** [comment]
A simple 'download my data' option would indeed be nice while data persistence is still shaky and providing backwards compatibility is too much effort. Downloaded data may have to be converted manually when we change formats, but the format is very simple now.

At the moment you can open the browser console and call `store.getState().storage` to see your stored docs and links. So you could save it as a file with:
`window.open( "data:text/json;charset=utf-8," + escape(JSON.stringify(store.getState().storage)))`

Loading data from a json file would require an extra action&reducer..

**Comment by BigBlueHat** [comment]
@Treora give this a look:
https://github.com/nolanlawson/pouchdb-replication-stream#pouchdb-replication-stream

**Comment by Treora** [comment]
I added some simple functions to the global scope:
- `dumpStorage()` gives the json file.
- `importFromDump()` prompts you for the json and adds it to the storage.
- `deleteAllDocs()` clears the storage.

**Comment by Treora** [comment]
For some unknown reason `dumpStorage()` nowadays opens a blank tab for me in Chromium, and a tab that directly closes itself in Firefox. I now run this in my browser console to get the dump in the current window (replacing the interface):
```
document.body.innerHTML = '<pre id="pre"></pre>'; pre.innerText = JSON.stringify(store.getState().storage, null, 2); null
```

(the `null` at the end prevents it from printing the dump in the console, which is slow)

I guess I am the only person still using this demo for practical purposes (understandably, it needs more love), so I won't bother changing the code now, but sharing this fix here anyway.

*Update: I fixed it anyhow, in commit ffcf15e. It now starts a download of the json file, much better than copying&pasting it each time.*
